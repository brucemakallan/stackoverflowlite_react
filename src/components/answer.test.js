import React from "react";
import { shallow } from "enzyme";
import Answer from "./Answer";

describe("<Answer />", () => {
  let answer, answerComponent;
  beforeEach(() => {
    answer = {
      id: 1,
      question_id: 1,
      answer: "A storage construct whose value can change",
      date_posted: "Fri, 03 Aug 2018 00:00:00 GMT"
    };
    answerComponent = shallow(<Answer answer={answer} />);
  });

  it("should render correctly", () => {
    expect(answerComponent).toMatchSnapshot();
  });

  it("should contain the correct answer", () => {
    expect(answerComponent.find("#answer_answer").text()).toEqual(
      answer.answer
    );
  });

  it("should contain the correct date posted", () => {
    expect(answerComponent.find("#answer_date_posted").text()).toEqual(
      answer.date_posted
    );
  });
});
