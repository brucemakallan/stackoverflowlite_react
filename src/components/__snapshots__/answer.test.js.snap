// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Answer /> should render correctly 1`] = `
<div
  className="answer-card m-2 p-2"
>
  <div
    className="row"
  >
    <div
      className="col col-10"
      id="answer_answer"
    >
      A storage construct whose value can change
    </div>
    <div
      className="col col-2 text-right"
    >
      <i
        className="fas fa-check green m-2"
      />
      <i
        className="fas fa-pencil-alt m-2 green"
      />
      <i
        className="fas fa-trash m-2 red"
      />
    </div>
    <div
      className="col col-12 my-2 small brown"
    >
      <i
        className="far fa-clock"
      />
      Â 
      <span
        id="answer_date_posted"
      >
        Fri, 03 Aug 2018 00:00:00 GMT
      </span>
    </div>
    <div
      className="col col-12 thumbs"
    >
      <i
        className="fas fa-thumbs-up"
      />
      <i
        className="fas fa-thumbs-down"
      />
      <span
        className="small brown"
      >
        # Votes
      </span>
    </div>
  </div>
</div>
`;
